python/util/templates.py

```diff
@@ -63,15 +66,19 @@ def generate_react_components(self):
         """ Generate javascript to render every component which has been marked
             used in this ComponentsDatabase, if any.
         """
+        view_embeds = public_views_embed.get(flask.request.endpoint, {})
```

Comment:
here's a good specific example of the problem of using flask in `sdk`.

suddenly if we want to use these functions outside of a flask request function, we can't. Or, even in the backend, which is a flask server, but `flask.request.endpoint` is not going to be the thing you expect here, if we are in the backend.

it's better to just provide the right embed values from the view directly.

Reply:
I can see your point If we are going to compile everything in the backend but for now this works well in the frontend.

We use use flask in 4 other files inside sdk

Reply:
the backend does pre-compile templates. i am not completely sure how much of the scripts it compiles though because the control flow is weird.

the usage of flask inside python/files.py is ok, because it it just pure file munging, and not interfering with the request flow. The usages of flask inside python/util/redirect.py, and python/entities.py, and the new one here are bad and should be removed.

Reply:
Well, thinking about it more it would be better not to depend at all on flask. But the first case is certainly much less problematic than the other ones.

Reply:
Can you think of a solution?